packages/ra-ui-materialui/src/input/ReferenceInput.tsx

```diff
@@ -226,14 +258,10 @@ export const ReferenceInputView = ({
         isRequired,
         label,
         resource,
-        meta: {
-            ...meta,
-            helperText: warning || false,
-        },
+        meta: finalMeta,
         source,
         choices,
         basePath,
-        onChange,
```

Comment:
why that change?

Reply:
Because user defined `onChange` is now handled in the `ReferenceInput` component through `useInput` (ensuring that it does not prevent final-form from working).

`onChange` is then injected back through the `input` prop inside `ReferenceInputView`